bool is_hooked
bool check1
bool check2
bool check3
bool check4
bool check5


